app/liquid_tags/forem_tag.rb

```diff
@@ -0,0 +1,52 @@
+module ForemTag
+  REGISTRY_REGEXP = %r{#{URL.url}/\w+(/[\w-]+)?(/[\w-]+)?}
+  USER_ORG_REGEXP = %r{#{URL.url}/(?<name>[\w-]+)/?$}
+  POST_PODCAST_REGEXP = %r{#{URL.url}/(?<podcast>[\w-]+)/[\w-]+/?}
+  COMBINED_REGEXP = [USER_ORG_REGEXP, POST_PODCAST_REGEXP].freeze
+
+  def self.new(tag_name, input, parse_context)
+    link = process_input(input)
+    klass = determine_klass(link)
+    raise StandardError, "Invalid #{Settings::Community.community_name} URL." unless klass
+
+    klass.__send__(:new, tag_name, link, parse_context)
+  end
+
+  def self.process_input(input)
+    stripped_input = ActionController::Base.helpers.strip_tags(input).strip
+    CGI.unescape_html(stripped_input)
+  end
+
+  def self.determine_klass(link)
+    return ListingTag if link.include?("#{URL.url}/listings/")
+    return TagTag if link.include?("#{URL.url}/t/")
```

Comment:
Do these need to be `include?` or is `start_with?` still accurate?

```suggestion
    return ListingTag if link.start_with?("#{URL.url}/listings/")
    return TagTag if link.start_with?("#{URL.url}/t/")
```